src/map/map.ts

```diff
@@ -128,38 +128,89 @@ function arrayMaxMin(values: number[]): {max: number, min: number} {
     return {max, min};
 }
 
-/** HTML element holding settings for a given axis (x, y, z, color) */
-class AxisSetting {
+interface AxisPresets {    
     /// Which property should we use for this axis
-    public property: HTMLSetting<'string'>;
+    property?: string;
     /// Which scale (linear/log) should we use
-    public scale: HTMLSetting<'string'>;
+    scale?: string;
     /// The minimal value for this axis
-    public min: HTMLSetting<'number'>;
+    min?: number;
     /// The maximal value for this axis
-    public max: HTMLSetting<'number'>;
+    max?: number;
+}
+
+const AxisDefaults = {
+     property : '',
+     scale : 'linear',
+     min : 0,
+     max: 0,
+}
 
-    constructor(properties: string[], initial: string) {
+/** HTML element holding settings for a given axis (x, y, z, color) */
+class AxisSettings {
+    public property!: HTMLSetting<'string'>;
+    public scale!: HTMLSetting<'string'>;
+    public min!: HTMLSetting<'number'>;
+    public max!: HTMLSetting<'number'>;
+
+    private _properties: string [];
+    
+    constructor(properties: string[], presets?: AxisPresets) {
+        this._properties = properties;
+        
+        // creates settings
         this.max = new HTMLSetting('number', 0);
         this.min = new HTMLSetting('number', 0);
-
-        this.property = new HTMLSetting('string', initial);
+        this.property = new HTMLSetting('string', '');
+        this.scale = new HTMLSetting('string', '');
+                
+        this.applyPresets(presets);
```

Comment:
I would apply the presets after setting `validate`, or else bad preset will not be caught

Reply:
Had put it there so that there was an initial value already set for the validate function to pick up. I now streamlined the whole validation thing